extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/cors/CORSFilter.java

```diff
@@ -140,11 +140,11 @@ public void handle(RoutingContext event) {
 
             //for both normal and preflight requests we need to check the origin
             boolean allowsOrigin = wildcardOrigin;
+            boolean originMatches = corsConfig.origins.isPresent() &&
```

Comment:
This check is only expected to be done only if the allowed origins are not configured to `*` (wildcard), the origin is assumed to be always matching otherwise